src/wp-includes/blocks.php

```diff
@@ -331,8 +332,9 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 	}
 
 	// Try to get metadata from the static cache for core blocks.
-	$metadata = false;
-	if ( str_starts_with( $file_or_folder, ABSPATH . WPINC ) ) {
+	$metadata      = false;
+	$is_core_block = str_starts_with( $file_or_folder, ABSPATH . WPINC );
```

Comment:
We're starting to build up lots of places across this file where we are trying to determine if something is a core block before doing extra file checks. Would be nice to consolidate all of this logic into a single helper function.